lib/player_ai_minimax.rb

```diff
@@ -0,0 +1,60 @@
+class PlayerAiMinimax
+  attr_reader :mark
+
+  def initialize(mark, rules)
+    @mark = mark
+    @rules = rules
```

Comment:
You can define `attr_reader`s as `private`. Here's one nice feature of using readers over instance variables throughout your class:

```
class Foo
  def initialize(bar, baz)
    @bar = bar
    @baz = baz
  end

  def get_bar
    @bat   # misspelled @bar instance variable
  end

  def get_baz
    bax   # misspelled baz reader method
  end
  
  private
  attr_reader :baz
end


f = Foo.new(1,2)
f.get_bar  # => nil.
           # It's unclear whether @bar didn't get set or if the instance variable was just misspelled somewhere

f.get_baz  # => NameError: undefined local variable or method `bax` for #<Foo>.
           # Ah! It's clear I misspelled `baz` somewhere
```

I also happen to think it looks cleaner than having instance variables everywhere...